File: gremlin-core/src/test/java/com/tinkerpop/gremlin/filter/RangeStepTest.java
package com.tinkerpop.gremlin.filter;

import com.tinkerpop.blueprints.pgm.Graph;
import com.tinkerpop.blueprints.pgm.Vertex;
import com.tinkerpop.blueprints.pgm.impls.tg.TinkerGraphFactory;
import com.tinkerpop.gremlin.pipes.GremlinPipeline;
import com.tinkerpop.gremlin.test.UtilitiesTest;

/**
 * @author Marko A. Rodriguez (http://markorodriguez.com)
 */
public class RangeStepTest extends com.tinkerpop.gremlin.test.filter.RangeStepTest {

    Graph g = TinkerGraphFactory.createTinkerGraph();

    public void testCompliance() {
        UtilitiesTest.testCompliance(this.getClass());
    }

    public void test_g_v1_out_rangeX0_1X() {
        super.test_g_v1_out_rangeX0_1X(new GremlinPipeline<Vertex, Vertex>(g.getVertex(1)).out().range(0, 1));
    }

}

File: gremlin-scala/src/test/scala/com/tinkerpop/gremlin/scala/filter/RangeStepTest.scala
package com.tinkerpop.gremlin.scala.filter

import com.tinkerpop.blueprints.pgm.impls.tg.TinkerGraphFactory
import com.tinkerpop.gremlin.scala._
import com.tinkerpop.gremlin.test.ComplianceTest

/**
 * @author Marko A. Rodriguez (http://markorodriguez.com)
 */

class RangeStepTest extends com.tinkerpop.gremlin.test.filter.RangeStepTest {

  val g = TinkerGraphFactory.createTinkerGraph();

  override def testCompliance() {
    ComplianceTest.testCompliance(this.getClass);
  }

  def test_g_v1_out_rangeX0_1X() {
    super.test_g_v1_out_rangeX0_1X(g.v(1).out.range(0, 1));
  }
}